src/polyhedralGravity/model/SplitNode.h

```diff
@@ -0,0 +1,79 @@
+#pragma once
+
+#include <memory>
+
+#include "KdDefinitions.h"
+#include "TreeNode.h"
+
+class SplitNode final : public TreeNode {
+
+    /**
+     * The SplitNode that contains the bounding box closer to the origin with respect to the split plane
+     */
+    std::unique_ptr<TreeNode> _lesser;
```

Comment:
What's the reason you use `unique_ptr` for everything?

(This is not a suggestion to change, but just what are the reasons?)

Reply:
I had a prof who really liked unique_ptr to be honest there's no real reason. I'll revisit the usages and reconsider if they are actually necessary